scene/animation/animation_node_extension.h

```diff
@@ -0,0 +1,189 @@
+/**************************************************************************/
+/*  animation_node_extension.h                                            */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef ANIMATION_NODE_EXTENSION_H
+#define ANIMATION_NODE_EXTENSION_H
+
+#include "scene/animation/animation_tree.h"
+
+class NodeTimeInfoRC : public RefCounted {
```

Comment:
What does RC mean?

Reply:
We use full names rather than short ones in the godot engine C++ naming style.

Reply:
Ref Counted, though I spent 0 time coming up with a good name for this, open to suggestions!

Reply:
Why does it need this clarification? Does it collide with anything? Could it be confused for something?

Reply:
Typically, RC in godot engine means a virtual function override in bindings, which means it has returns and is constant, which has nothing to do with this.

Reply:
`NodeTimeInfoExtension` and `PlaybackInfoExtension` would extend the naming convention we're creating.

Reply:
Yeah thats the thing @AThousandShips , both of these are really just  wrappers around `AnimationNode::NodeTimeInfo` and `AnimationMixer::PlaybackInfo`, so that extensions/GDScripts can override the `AnimationNode::_process` function. So I didn't want to name them the same, but I don't love the "RC" suffix either

Reply:
In summary, `NodeTimeInfoExtension` and `PlaybackInfoExtension` would extend the naming convention we're creating, and I'd like to use this naming; unless you have other better naming ideas, please try them!